### DISCUSSION TITLE
Terraform Plugin Protocol and Licensing

### DISCUSSION THREAD

**Discussion by danieladams456** [opentofu/opentofu#538] [General]
Hi!  This is mainly a curiosity question on the longer-term potential divergence between OpenTofu and Terraform.  I'm not an expert on software licenses, so maybe this isn't a concern.

Is the [Terraform Plugin Protocol](https://developer.hashicorp.com/terraform/plugin/terraform-plugin-protocol) definition part of the Terraform repo and under the new BSL license?  What happens if Hashicorp "evolves" the plugin protocol in a non-backwards compatible way?  Can OpenTofu adopt the new interface?  Does this put OpenTofu in the same situation Java has been in over the years?  Does the [Google v. Oracle](https://www.gibsondunn.com/supreme-court-holds-that-googles-use-of-oracles-java-software-interface-is-fair-use/) case help?

**Comment by apparentlymart** [comment]
The protocol definition is also included in the MPL-licenced [terraform-plugin-go](https://github.com/hashicorp/terraform-plugin-go) library that most provider implementations directly or indirectly depend on. That will presumably remain true as long as HashiCorp intends to continue allowing open source provider plugins.

We've already adopted newer versions of the protocol in OpenTofu based on content in that repository, so I think that's a sufficient answer to this question. :grinning:
